File: smallest_difference/smallest_difference.py
# O(n * m) time | O(1) space
def smallest_difference(array_one, array_two):
  closest_pair = [array_one[0] - array_two[0]]
  closest_to_zero = abs(array_one[0] - array_two[0])
  for i in range(len(array_one)):
    for j in range(len(array_two)):
      current_value = abs(array_one[i] - array_two[j])
      if current_value < closest_to_zero:
        closest_to_zero = current_value
        closest_pair = [array_one[i], array_two[j]]
  return closest_pair

print(smallest_difference([-1, 5, 10, 20, 28, 3], [26, 134, 135, 15, 17]))

File: smallest_difference/smallest_difference.go
package main

import (
	"fmt"
	"math"
)

func smallestDifference(arrayOne, arrayTwo []int) []int {
	closestPair := []int{arrayOne[0], arrayTwo[0]}
	closestDiff := math.Abs(float64(arrayOne[0] - arrayTwo[0]))

	for _, a := range arrayOne {
		for _, b := range arrayTwo {
			currentDiff := math.Abs(float64(a - b))
			if currentDiff < closestDiff {
				closestDiff = currentDiff
				closestPair = []int{a, b}
			}
		}
	}
	return closestPair
}

func main() {
	fmt.Println(smallestDifference([]int{-1, 5, 10, 20, 28, 3}, []int{26, 134, 135, 15, 17}))
}
